spring-integration-core/src/main/java/org/springframework/integration/transaction/DefaultTransactionSynchronizationFactory.java

```diff
@@ -26,10 +26,13 @@
  * Default implementation of {@link TransactionSynchronizationFactory} which takes an instance of
  * {@link TransactionSynchronizationProcessor} allowing you to create a {@link TransactionSynchronization}
  * using {{@link #create(Object)} method.
+ * <p>
+ * If resource under the provided {@code key} is already registered, the factory returns {@code null}.
  *
```

Comment:
We can, but with that the `TransactionSynchronizationFactory` stops to be `@FunctionalInterface`. And there is no guarantee that custom implementation won't return `null` by some reason.